gems/sorbet-runtime/test/types/runtime_levels.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+require_relative '../test_helper'
+
+module Opus::Types::Test
+  class RuntimeLevelsTest < Critic::Unit::UnitTest
+    describe 'set_enable_checking_in_tests_from_environment' do
+      before do
+        @orig_wrapped_tests_with_validation = T::Private::RuntimeLevels.instance_variable_get(:@wrapped_tests_with_validation)
+        @orig_check_tests = T::Private::RuntimeLevels.instance_variable_get(:@check_tests)
+
+        # Within these specs pretend we haven't yet read this value
+        T::Private::RuntimeLevels.instance_variable_set(:@wrapped_tests_with_validation, false)
+      end
+
+      after do
+        T::Private::RuntimeLevels.instance_variable_set(:@check_tests, @orig_check_tests)
+        T::Private::RuntimeLevels.instance_variable_set(:@wrapped_tests_with_validation, @orig_wrapped_tests_with_validation)
+      end
+
+      it 'does not change check_tests' do
+        # Reaching into a private method for testing purposes
+        T::Private::RuntimeLevels.send(:set_enable_checking_in_tests_from_environment)
+
+        assert_equal(T::Private::RuntimeLevels.check_tests?, false)
+      end
+
+      describe 'when SORBET_RUNTIME_ENABLE_CHECKING_IN_TESTS env variable is set' do
+        before do
+          @orig_sorbet_runtime_enable_checking_in_tests = ENV['SORBET_RUNTIME_ENABLE_CHECKING_IN_TESTS']
+          ENV['SORBET_RUNTIME_ENABLE_CHECKING_IN_TESTS'] = '1'
+        end
+
+        after do
+          ENV['SORBET_RUNTIME_ENABLE_CHECKING_IN_TESTS'] = @orig_sorbet_runtime_enable_checking_in_tests
+        end
+
+        it 'updates check_tests' do
+          # Reaching into a private method for testing purposes
+          T::Private::RuntimeLevels.send(:set_enable_checking_in_tests_from_environment)
+
+          assert_equal(T::Private::RuntimeLevels.check_tests?, true)
+        end
+      end
+    end
+
+    describe 'set_default_checked_level_from_environment' do
+      before do
+        @orig_has_read_default_checked_level = T::Private::RuntimeLevels.instance_variable_get(:@has_read_default_checked_level)
+        @orig_default_checked_level = T::Private::RuntimeLevels.instance_variable_get(:@default_checked_level)
+
+        # Within these specs pretend we haven't yet read this value
+        T::Private::RuntimeLevels.instance_variable_set(:@has_read_default_checked_level, false)
+      end
+
+      after do
+        T::Private::RuntimeLevels.instance_variable_set(:@default_checked_level, @orig_default_checked_level)
+        T::Private::RuntimeLevels.instance_variable_set(:@has_read_default_checked_level, @orig_has_read_default_checked_level)
+      end
+
+      it 'does not change default_typed_level' do
+        # Reaching into a private method for testing purposes
+        T::Private::RuntimeLevels.send(:set_default_checked_level_from_environment)
+
+        assert_equal(T::Private::RuntimeLevels.default_checked_level, :always)
+      end
```

Comment:
Same thing here.

Reply:
This actually shouldn't need to update that. `set_default_checked_level_from_environment ` only reads from the environment variable, doesn't mutate it, and then calls `default_checked_level=` if the environment variable is set.

Reply:
```
froydnj@pyro:~/src/sorbet/gems/sorbet-runtime$ env SORBET_RUNTIME_DEFAULT_CHECKED_LEVEL=never PATH=$PATH:/home/froydnj/srb-ruby/bin /home/froydnj/srb-ruby/bin/bundle exec rake test
/home/froydnj/src/sorbet/gems/sorbet-runtime/test/types/method_validation.rb:573: warning: key :_ is duplicated and overwritten on line 573
Run options: --seed 48106

# Running:

....................................................................................................................................................................................................................................F.......................................................................................................................................................................................................................................................F...............................................................................................................................................................................................................................................................S...................................................................................................................

Finished in 0.455613s, 1861.2274 runs/s, 5774.6337 assertions/s.

  1) Failure:
set_default_checked_level_from_environment#test_0001_does not change default_typed_level [/home/froydnj/src/sorbet/gems/sorbet-runtime/test/types/runtime_levels.rb:64]:
Expected: :never
  Actual: :always

  2) Failure:
Opus::Types::Test::TopLevel#test_0001_works with top level sigs [/home/froydnj/src/sorbet/gems/sorbet-runtime/test/types/top_level.rb:9]:
--- expected
+++ actual
@@ -1,5 +1,5 @@
 "called main#foo
-Expected type Integer, got type String with value \"nope\"
+foo() did not raise a TypeError!
 called main.bar
-Expected type Symbol, got type Float with value 0.0
+bar() did not raise a TypeError!
 "


848 runs, 2631 assertions, 2 failures, 0 errors, 1 skips

You have skipped tests. Run with --verbose for details.
```

It's possible that both of these failures (and the one above) should be fixed at a higher level than the individual tests.